mitiq/calibration/calibrator.py

```diff
@@ -221,15 +246,33 @@ def expval_executor(circuit: cirq.Circuit) -> float:
 
 def execute_with_mitigation(
     circuit: QPROGRAM,
-    calibrator: Calibrator,
-    bitstring: str,
+    executor: Union[Executor, Callable[[QPROGRAM], QuantumResult]],
     observable: Optional[Observable] = None,
+    *,
+    calibrator: Calibrator,
 ) -> QuantumResult:
+    """Estimates the error-mitigated expectation value associated to the
+    input circuit, via the application of the best mitigation strategy, as
+    determined by calibration.
+
+
+    Args:
+        circuit: The input circuit to execute.
+        executor: A Mitiq executor that executes a circuit and returns the
+            unmitigated ``QuantumResult`` (e.g. an expectation value).
+        observable: Observable to compute the expectation value of. If
+            ``None``, the ``executor`` must return an expectation value.
+            Otherwise, the ``QuantumResult`` returned by ``executor`` is used
+            to compute the expectation of the observable.
+        calibrator: ``Calibrator`` object with which to determine the error
+            mitigation strategy to execute the circuit.
+
+    Returns:
```

Comment:
A very nice-to-have suggestion: a full_output option that when True, execute_with_mitigation also returns the best_strategy, improvement_factor, etc.
Could save for future milestone, not blocking from my POV.

Reply:
Opened  #1683 for tracking this suggestion.